### ISSUE TITLE
Android API Range Support?

### ISSUE THREAD

**Issue by bleege** [mapbox/mapbox-gl-native#605]
**tl;dr - How to make Android GL as widely available / usable as possible?**

In order to make sure that as many people / developers as possible can use Android GL in a developer resource realistic way we're going to have to think about Android API ranges.  Google says that OpenGL 2.0 support [is available at API 8](%28http://developer.android.com/guide/topics/graphics/opengl.html%29) and above.  The good news is that there is a [minuscule number](https://developer.android.com/about/dashboards/index.html) of those devices being used these days.  That said those same numbers also show that there's still **10% of all Android devices in the wild** are running API 10.  Yikes!

The good news is that the vast majority seem to be in the Jelly Bean range (API 16, 17, and 18).  Currently it looks like Android GL is using [API 14](https://github.com/mapbox/mapbox-gl-native/blob/android-mason/android/java/app/build.gradle#L9) as a floor.  What do you all think about using API 10 as a floor (for the time being until those numbers drop) while always adjusting the ceiling to the latest and greatest Android API (currently API 21)?

Finally, I can never keep track of all the dessert names and API numbers that Google puts out (I just get hungry thinking about it though.).  Thankfully Google keeps a list of 'em all for us: [Android Code Names And API Names](https://source.android.com/source/build-numbers.html).

![jbstatue2](https://cloud.githubusercontent.com/assets/989928/5059114/54b102da-6cd4-11e4-974d-697ee4254af4.jpg)

/cc
@incanus @ljbade @springmeyer 

Also see #604
